docs/concepts/json_schema.md

```diff
@@ -1029,6 +1076,93 @@ print(json.dumps(TypeAdapter(Person).json_schema(), indent=2))
 ```
 
 
+### Using `field_title_generator`
+
+The `field_title_generator` parameter can be used to dynamically generate the title for a field based on its name.
+This is similar to the field level `field_title_generator`, but the `ConfigDict` option will be applied to all fields of the class.
+
+See the following example:
+
+```py
+import json
+
+from pydantic import BaseModel, ConfigDict
+
+
+class Person(BaseModel):
+    model_config = ConfigDict(field_title_generator=lambda m: m.upper())
+    name: str
+    age: int
+
+
+print(json.dumps(Person.model_json_schema(), indent=2))
+"""
+{
+  "properties": {
+    "name": {
+      "title": "NAME",
+      "type": "string"
+    },
+    "age": {
+      "title": "AGE",
+      "type": "integer"
+    }
+  },
+  "required": [
+    "name",
+    "age"
+  ],
+  "title": "Person",
+  "type": "object"
+}
+"""
+```
+
+### Using `class_title_generator`
+
+The `class_title_generator` config option is similar to the `field_title_generator` option, but it applies to the title of the class itself.
+
+See the following example:
+
+```py
+import json
+
+from pydantic import BaseModel, ConfigDict
+
+
+def make_title(field_name: str) -> str:
```

Comment:
again I would suggest we pass the `cls` itself, as well as the current name.

Reply:
Sure. 
What type do you reckon the `cls` parameter be annotated as? `Type[BaseModel]`?